app/views/users/edit.html.erb

```diff
@@ -0,0 +1,23 @@
+
+<%= form_for @user do |f|%>
+  <%= f.label :username, "Username:"%>
+  <%= f.text_field :username, value: "u3" %><br><br>
+  <%= f.label :password, "Password:"%>
+  <%= f.password_field :password%><br><br>
+  <%= f.label :email, "Email:"%>
+  <%= f.text_field :email, placeholder: "example@example.com"%><br><br>
+
+  <%= f.submit "Update"%>
+<%end%>
+
+<% if @user_temp.errors.any? %>
```

Comment:
`<% if @user_temp.errors.any? %>` means you are creating a new user entirely and checking if all fields are present. Use `<% if @user.errors.any? %>` instead, that way, you validate the exact user that you are editing.

Also, don't forget to replace `@user_temp` with `@user` in other places too.

Reply:
Thanks for the suggestion. It's been fixed